src/main/scala/wdl4s/Scope.scala

```diff
@@ -183,59 +183,65 @@ trait Scope {
+          Failure(VariableLookupException(s"Failed to evaluate scatter expression (${scatter.collection.toWdlString})", ex))
         case (_, None) =>
-          throw new VariableLookupException(s"Could not find a shard for scatter block with expression (${scatter.collection.toWdlString})")
+          Failure(VariableLookupException(s"Could not find a shard for scatter block with expression (${scatter.collection.toWdlString})"))
       }
     }
 
-    def handleDeclarationEvaluation(declaration: DeclarationInterface): Option[WdlValue] = {
-      for {
-        expression <- declaration.expression
-        parentLookup = declaration.parent.map(_.lookupFunction(knownInputs, wdlFunctions, outputResolver, shards)).getOrElse(NoLookup)
-        value = expression.evaluate(parentLookup, wdlFunctions).get
-      } yield value
+    def handleDeclarationEvaluation(declaration: DeclarationInterface): Try[WdlValue] = {
+      declaration.expression match {
+        case Some(e) =>
+          val parentLookup = declaration.parent.map(_.lookupFunction(knownInputs, wdlFunctions, outputResolver, shards)).getOrElse(NoLookup)
+          e.evaluate(parentLookup, wdlFunctions)
+        case None =>
+          Failure(VariableLookupException(s"Declaration ${declaration.fullyQualifiedName} does not have an expression"))
+      }
     }
 
-    def handleCallEvaluation(call: Call): Option[WdlValue] = {
+    def handleCallEvaluation(call: Call): Try[WdlValue] = {
       this match {
           // Only use the shard number if the call is inside the scatter
         case s: Scatter if children.contains(call) => shards.get(s) map { shard =>
-          outputResolver(call, Option(shard)) getOrElse {
-            throw new VariableLookupException(s"Could not find outputs for call ${call.fullyQualifiedName} at shard $shard")
-          }
-        } orElse {
-          throw new VariableLookupException(s"Could not find a shard for scatter block with expression (${s.collection.toWdlString})")
+          outputResolver(call, Option(shard))
+        } getOrElse {
+          Failure(VariableLookupException(s"Could not find a shard for scatter block with expression (${s.collection.toWdlString})"))
         }
-        case _ => outputResolver(call, None).toOption
+        case _ => outputResolver(call, None)
       }
     }
 
     def lookup(name: String): WdlValue = {
-      val scopeResolvedValue = resolveVariable(name, relativeTo) flatMap {
+      val scopeResolvedValue = resolveVariable(name, relativeTo) map {
         // First check if the variable has been provided in the known values
-        case scope if knownInputs.contains(scope.fullyQualifiedName) => knownInputs.get(scope.fullyQualifiedName)
+        case scope if knownInputs.contains(scope.fullyQualifiedName) => 
+          knownInputs.get(scope.fullyQualifiedName) map Success.apply getOrElse {
```

Comment:
I'm not suggesting one be created, just that I thought we had one somewhere. As to how it would work, dunno - my assumption would be some stupid exception.